tinygrad/codegen/linearizer.py

```diff
@@ -607,10 +607,13 @@ def limit_global_dims(self, limit):
     # sometimes, there's more dimensions than len(self.lang.gid).
     # compact all the dimensions into the first
     # NOTE: this might make multiview shapetrackers
-    if limit and (self.first_reduce-self.local_dims) > limit:
-      num_to_merge = ((self.first_reduce-self.local_dims) - limit)+1
-      self.reshape_and_permute(lambda x: (prod(x[0:num_to_merge]),)+x[num_to_merge:], None)
-      if DEBUG >= 3: print("reshaped to", self.full_shape, "due to too many global dimensions")
+    if limit:  
+      if (self.first_reduce-self.local_dims) > limit:
+        num_to_merge = ((self.first_reduce-self.local_dims) - limit)+1
+        self.reshape_and_permute(lambda x: (prod(x[0:num_to_merge]),)+x[num_to_merge:], None)
+        if DEBUG >= 3: print("reshaped to", self.full_shape, "due to too many global dimensions")
+      global_dims = self.first_reduce-self.local_dims
```

Comment:
I can probably add a `if Device.DEFAULT == "CUDA":` to pass CI but that doesn't feel right. Will look into why the additional reshape broke opencl.